src/Internals/ThriftAttributesParser.cs

```diff
@@ -16,91 +16,95 @@ namespace ThriftSharp.Internals
+            if (attr.IsRequired)
             {
-                if( attr.DefaultValue != null )
+                if (attr.DefaultValue != null)
                 {
-                    throw ThriftParsingException.DefaultValueOnRequiredField( propertyInfo );
+                    throw ThriftParsingException.DefaultValueOnRequiredField(propertyInfo);
                 }
-                if( nullableType != null )
+                if (nullableType != null)
                 {
-                    throw ThriftParsingException.RequiredNullableField( propertyInfo );
+                    throw ThriftParsingException.RequiredNullableField(propertyInfo);
                 }
             }
             else
             {
-                if( attr.DefaultValue == null && propertyTypeInfo.IsValueType && nullableType == null )
+                if (attr.DefaultValue == null && propertyTypeInfo.IsValueType && nullableType == null)
                 {
-                    throw ThriftParsingException.OptionalValueField( propertyInfo );
+                    throw ThriftParsingException.OptionalValueField(propertyInfo);
                 }
             }
 
-            if( attr.DefaultValue != null )
+            if (attr.DefaultValue != null)
             {
-                if( attr.Converter == null )
+                if (attr.Converter == null)
                 {
-                    if( attr.DefaultValue.GetType() != ( nullableType ?? propertyInfo.PropertyType ) )
+                    if (attr.DefaultValue.GetType() != (nullableType ?? propertyInfo.PropertyType))
                     {
-                        throw ThriftParsingException.DefaultValueOfWrongType( propertyInfo );
+                        throw ThriftParsingException.DefaultValueOfWrongType(propertyInfo);
                     }
                 }
                 else
                 {
-                    if( attr.DefaultValue.GetType() != attr.ThriftConverter.FromType )
+                    if (attr.DefaultValue.GetType() != attr.ThriftConverter.FromType)
                     {
-                        throw ThriftParsingException.DefaultValueOfWrongType( propertyInfo );
+                        throw ThriftParsingException.DefaultValueOfWrongType(propertyInfo);
                     }
                 }
             }
 
-            return new ThriftField( attr.Id, attr.Name, attr.IsRequired, attr.DefaultValue, attr.ThriftConverter, propertyInfo );
+            return new ThriftField(attr.Id, attr.Name, attr.IsRequired, attr.DefaultValue, attr.ThriftConverter,
+                propertyInfo);
         }
 
         /// <summary>
         /// Parses a Thrift struct from the specified TypeInfo.
         /// </summary>
-        public static ThriftStruct ParseStruct( TypeInfo typeInfo )
+        public static ThriftStruct ParseStruct(TypeInfo typeInfo)
         {
-            if( !_knownStructs.ContainsKey( typeInfo ) )
+            if (_knownStructs.ContainsKey(typeInfo))

```

Comment:
I think you can use _knownStructs.GetOrAdd(key, valueFunc) here.  It will do what your doing -- check if the key exists, execute valueFunc, check if the key exists in the dict again before adding, and then return whatever's in the dictionary

https://docs.microsoft.com/en-us/dotnet/api/system.collections.concurrent.concurrentdictionary-2.getoradd?view=netframework-4.7.1#System_Collections_Concurrent_ConcurrentDictionary_2_GetOrAdd__0_System_Func__0__1__